### PR TITLE
Add 3D tile 1D with micro-arch functions

### PR DESCRIPTION
**Pull request by ngzhian**

### CODE REVIEW

**Comment by ngzhian** [comment]
Depends on #31

**Comment by ngzhian** [comment]
Ned to rebase, 1 sec.

**Comment by ngzhian** [comment]
Done, ready for merge.

File: include/pthreadpool.h [hunk 1]
```diff
@@ -474,7 +475,7 @@ void pthreadpool_parallelize_2d_tile_2d_with_uarch(
 	uint32_t flags);
 
 /**
- * Process items on a 3D grid.
+ * Process items on a 3D grid with the specified maximum tile size along the
```
Comment (Maratyszcza) [inline include/pthreadpool.h:478]: The updated comment is wrong

**Review by Maratyszcza — CHANGES_REQUESTED** [review]

File: include/pthreadpool.h [hunk 2]
```diff
@@ -553,6 +554,64 @@ void pthreadpool_parallelize_3d_tile_1d(
 	size_t tile_k,
 	uint32_t flags);
 
+/**
+ * Process items on a 3D grid.
+ * last grid dimension using a microarchitecture-aware task function.
```
Comment (Maratyszcza) [inline include/pthreadpool.h:559]: Comment is wrong

**Review by Maratyszcza — CHANGES_REQUESTED** [review]

File: include/pthreadpool.h (hunk 2 above)
Comment (ngzhian) [reply include/pthreadpool.h:559]: Good catch, thanks

**Review by Maratyszcza — APPROVED** [review]

**Pull request merged**